shell/detail/catalog.cattle.io.app.vue

```diff
@@ -30,9 +31,15 @@ export default {
   },
 
   async fetch() {
-    await this.$store.dispatch('catalog/load');
+    const promises = {
+      catalog:       this.$store.dispatch('catalog/load'),
+      allOperations: this.$store.dispatch('cluster/findAll', { type: CATALOG.OPERATION }),
+      secrets:       this.value.fetchValues(true),
```

Comment:
@richard-cox Will the page now fail to render if the user does not have access to secrets?

Reply:
From this line no (given https://github.com/rancher/dashboard/pull/12317/files#diff-19610955367613fc802e3d3377ad5c361325e01607dbbd4b5c3feda01b55c860R334). Later on though i do throw some exceptions if they try to access app getters that depend on the secret (https://github.com/rancher/dashboard/pull/12317/files#diff-19610955367613fc802e3d3377ad5c361325e01607dbbd4b5c3feda01b55c860R360).

This was roughly defined in the RFC though. Users that manage apps should..
- be able to see the namespace of the secret given it's the same namespace as the app
- be able to _create_ secrets in the namespace given that the operator already does this on behalf of the user
- have the get / delete secrets permission
  - RFC states `The permission is a new requirement, does not come by default`.
  - however... i'd tried this out as a `standard-user` who was a project member of a project containing a namespace with an app in.... and the app could be fetched ok. the secret however was not updated (taking to @diogoasouza)